nmpolicy/internal/parser/parser.go

```diff
@@ -31,5 +34,51 @@ func New() Parser {
 }
 
 func (p Parser) Parse(tokens []lexer.Token) (ast.Node, error) {
+	if reflect.DeepEqual(tokens, []lexer.Token{
+		{Position: 0, Type: lexer.IDENTITY, Literal: "routes"},
```

Comment:
You could use variables to place the literal values. Then use them also in the ast tree, showing the relation and even avoiding the `strPtr` function.

But as this code is intermediate anyway, I will leave it to you to decide if it is worth talking care of.